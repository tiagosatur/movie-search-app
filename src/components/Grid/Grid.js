import React from 'react';
import styled from 'styled-components';
import { rem } from '../../style'

const Grid = ({children}) => <StyledGrid className='grid'>{children}</StyledGrid>

const StyledGrid = styled.div`
    height: 100%;
    padding: ${rem(16)};
    @media(min-width: ${rem(768)}) {
        padding: 0;
        width: 80%;
        margin: 0 auto;
    }
`;

export default Grid;