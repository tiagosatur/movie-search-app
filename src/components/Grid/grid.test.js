import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Grid from './Grid';

describe('Grid tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Grid> 
          <div>Some Grid content</div>
      </Grid>
    ).dive();
  });
  
  it('should render the Grid element', () => {
    expect(wrapper.find('.grid').exists()).toBe(true);
  });

  it('should call Grid snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})