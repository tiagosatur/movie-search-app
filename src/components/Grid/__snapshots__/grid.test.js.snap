// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid tests should call Grid snapshot correctly 1`] = `
<div
  className="grid sc-bdVaJa kcCvpD"
>
  <div>
    Some Grid content
  </div>
</div>
`;
